src/misc_early.rs

```diff
@@ -113,4 +125,15 @@ impl EarlyLintPass for MiscEarly {
             }
         }
     }
+
+    fn check_expr(&mut self, cx: &EarlyContext, expr: &Expr) {
+        if let ExprKind::Call(ref paren, _) = expr.node {
+            if let ExprKind::Paren(ref closure) = paren.node {
+                if let ExprKind::Closure(_, _, _) = closure.node {
+                    cx.span_lint(CLOSURE_CALLED_EARLY, expr.span, "Try not to call \
+                        a closure in the expression where it is declared.")
```

Comment:
Also don’t use `cx.span_lint` directly here, but one of Clippy’s [`span_*lint*` functions](https://github.com/Manishearth/rust-clippy/blob/master/src/utils/mod.rs#L462-L514), it will automatically add a link to the lint’s documentation.